Assets/Scripts/Checkpoint.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Animations;
+
+public class Checkpoint : MonoBehaviour {
+    [SerializeField]
+    public bool isActivated = false;
+   
+    public Animator anim;
+
+    //creates a variable that refers to the PlayerController
+    PlayerController player;
+
+    private void Start()
+    {
+        anim = this.GetComponent<Animator>();       
+    }
+
+    private void Update()
+    {
+        UpdateAnimation();
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.gameObject.tag == "Player")
+        {            
+            player = collision.GetComponent<PlayerController>();
+            player.SetCurrentCheckpoint(this);            
+        }
+    }
+
+    public void SetAsActivated(bool value)
```

Comment:
you could try making this a property and put these lines in the "setter" of the property.